include/drivers/pcie/pcie.h

```diff
@@ -35,6 +36,12 @@ typedef uint32_t pcie_bdf_t;
  */
 typedef uint32_t pcie_id_t;
 
+struct pci_mbar {
+    uintptr_t phys_addr;
+    size_t    size;
+    uint8_t   flags;
```

Comment:
actually I don't think caller will care of getting the flags. So you can remove this.
(on I/O these are fully useless for the caller, and on MEM I don't think we care about the prefetchable bit at all)

Unless knowing that it's an I/O address instead of a MEM is necessary for the caller? (then bool mem; would just do the job)

Reply:
Yes for me it is relevant to know whether the bar is I/O or MEM. I personally do not need the other flags, so I will change this to bool.

Reply:
I did change this to `bool mem`. The alternative would be to skip I/O Bars aswell.